In [2]:
import qcportal as ptl
import pandas as pd
import datetime
import time

from management import *

from lifecycle import DataSet

In [3]:
# connect without auth
# read only
client = ptl.FractalClient()

In [4]:
client

FractalClient(server_name='The MolSSI QCArchive Server', address='https://api.qcarchive.molssi.org:443/', username='None')

In [5]:
dataset = "OpenFF Gen 2 Torsion Set 1 Roche 2"

In [6]:
ds = client.get_collection("TorsionDriveDataset", dataset)

In [7]:
ds.list_specifications()

Description
Name                                                
default  Standard OpenFF torsiondrive specification.

In [9]:
results = defaultdict(dict)
for spec in ds.list_specifications().index.tolist():
    tdrs = get_torsiondrives(ds, spec, client)

    for status in ['COMPLETE', 'RUNNING', 'ERROR']:
        results[spec][status] =  len(
                [tdr for tdr in tdrs if tdr.status == status])

In [10]:
df_tdr = pd.DataFrame(results).transpose()

In [11]:
df_tdr.index.name = 'specification'

In [12]:
df_tdr

COMPLETE  ERROR  RUNNING
specification                          
default             136      1        5

In [14]:
print(df_tdr.to_markdown())

| specification   |   COMPLETE |   ERROR |   RUNNING |
|:----------------|-----------:|--------:|----------:|
| default         |        136 |       1 |         5 |


In [16]:
results = defaultdict(dict)
for spec in ds.list_specifications().index.tolist():
    opts = merge(
            get_torsiondrive_optimizations(ds, spec, client))

    for status in ['COMPLETE', 'INCOMPLETE', 'ERROR']:
        results[spec][status] =  len(
                [opt for opt in opts if opt.status == status])

df_tdr_opt = pd.DataFrame(results).transpose()
df_tdr_opt.index.name = 'specification'

In [17]:
df_tdr_opt

COMPLETE  ERROR  INCOMPLETE
specification                             
default            6966      4           5

In [18]:
from datetime import datetime

In [19]:
datehr = datetime.utcnow().strftime("%Y-%m-%d %Hhr UTC")

In [40]:
dataset_type = 'TorsionDriveDataset'
dataset_name = 'OpenFF Gen 2 Torsion Set 1 Roche 2'

meta = {'**Dataset Name**': dataset_name,
        '**Dataset Type**': dataset_type,
        '**UTC Date**': datehr}

meta = pd.DataFrame(pd.Series(meta, name=""))

In [43]:
comment = f"""
## Error Cycling Report

{meta.to_markdown()}

### `TorsionDriveRecord` current status

{df_tdr.to_markdown()}

### `OptimizationRecord` current status

{df_tdr_opt.to_markdown()}

"""


In [44]:
print(comment)


## Error Cycling Report

|                  |                                    |
|:-----------------|:-----------------------------------|
| **Dataset Name** | OpenFF Gen 2 Torsion Set 1 Roche 2 |
| **Dataset Type** | TorsionDriveDataset                |
| **UTC Date**     | 2020-07-09 23hr UTC                |

### `TorsionDriveRecord` current status

| specification   |   COMPLETE |   ERROR |   RUNNING |
|:----------------|-----------:|--------:|----------:|
| default         |        136 |       1 |         5 |

### `OptimizationRecord` current status

| specification   |   COMPLETE |   ERROR |   INCOMPLETE |
|:----------------|-----------:|--------:|-------------:|
| default         |       6966 |       4 |            5 |




## Error Cycling Report
| --- | --- |
| :------------- | -----------: |
| **Dataset Type** | TorsionDriveDataset |
| **Dataset Name** | OpenFF Gen 2 Torsion Set 1 Roche 2 |
| **UTC Date** | 2020-07-09 23hr UTC |

### `TorsionDriveRecord` current status

| specification   |   COMPLETE |   ERROR |   RUNNING |
|:----------------|-----------:|--------:|----------:|
| default         |        136 |       1 |         5 |

### `OptimizationRecord` current status

| specification   |   COMPLETE |   ERROR |   INCOMPLETE |
|:----------------|-----------:|--------:|-------------:|
| default         |       6966 |       4 |            5 |


## DataSet testing

In [12]:
from github import Github
from getpass import getpass

In [14]:
ghapi = Github(getpass())

 ········································


In [10]:
dst = DataSet('../submissions/2020-03-23-OpenFF-Gen-2-Torsion-Set-1-Roche-2', )

TypeError: __init__() missing 2 required positional arguments: 'pr' and 'ghapi'

We'll begin our live testing here.